Controller/ExceptionController.php

```diff
@@ -70,15 +70,11 @@ public function showAction(Request $request, $exception, DebugLoggerInterface $l
             $code = $this->getStatusCode($exception);
         } else {
             $code = $this->getStatusCodeFromThrowable($exception);
+            $exception = new \Exception($exception->getMessage(), $code);
```

Comment:
I am not convinced that this will work as expected since we will lose the full exception stack trace here.

Reply:
I'm currently still working on another fix. But testing this solution everything seems to work just fine.
The View class doesn't see to accept a TypeError Exception as a first argument. Converting the instance to a \Exception seemed was a first workaround !

Reply:
Or to start with something else: Do you see the same error when using the Symfony serializer? If not, could we hook into the JMS serializer and fix how the exception is handled?

Reply:
Well after tracing the the problem down to its roots, I found out that the problem lies in the serialization handled by jms_serializer when handling the view object.
jms_serializer, somehow, can't manage to serialize the \TypeError Exception passed to the View Class and throws the previously mentioned exception !
I'll be looking deeply into this problem later. I'll explore another way to solve this !

Reply:
I think to this fix but i'm not convinced too ... for the meme reason

Reply:
Maybe we can catch these cases in the `ExceptionHandler` class. I can try to look into it later today.

@faresk93 Do you have an application that allows to reproduce it or is the one from #2092 good enough?

Reply:
A test case should be to return a different type than the on expected from a method.

My case was an Entity Class getter returning a PersistentCollection with a return Type (php 7.3) being set to ArrayCollection (instead of just Collection) which throws a \TypeError exception !
(I'm not quite sure about the issue you mentioned)

Reply:
Can you check if #2105 fixes this?